applications/tari_validator_node/src/registration.rs

```diff
@@ -140,8 +148,17 @@ async fn start(
             Ok(event) = rx.recv() => {
                 match event {
                     EpochManagerEvent::EpochChanged(current_epoch) => {
-                        if let Ok(Some(next_registration_epoch)) = epoch_manager.next_registration_epoch().await {
-                            if current_epoch == next_registration_epoch {
+                        if let Ok(Some(last_registration_epoch)) = epoch_manager.last_registration_epoch().await {
+                            let mut base_node_client = GrpcBaseNodeClient::new(config.validator_node.base_node_grpc_address.unwrap_or_else(|| {
```

Comment:
Made an issue for this to be cleaned up in future: https://github.com/tari-project/tari-dan/issues/168

Reply:
Great ! Will probably tackle that one, as well :)